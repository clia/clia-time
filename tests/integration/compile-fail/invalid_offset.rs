use time::macros::offset;

fn main() {
    let _ = offset!(+24);
    let _ = offset!(+0:60);
    let _ = offset!(+0:00:60);
    let _ = offset!(0);
    let _ = offset!();
    let _ = offset!(+0a);
    let _ = offset!(+0:0a);
    let _ = offset!(+0:00:0a);
}
