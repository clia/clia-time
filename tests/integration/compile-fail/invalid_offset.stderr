error: invalid component: hour was 24
 --> $DIR/invalid_offset.rs:4:22
  |
4 |     let _ = offset!(+24);
  |                      ^^

error: invalid component: minute was 60
 --> $DIR/invalid_offset.rs:5:24
  |
5 |     let _ = offset!(+0:60);
  |                        ^^

error: invalid component: second was 60
 --> $DIR/invalid_offset.rs:6:27
  |
6 |     let _ = offset!(+0:00:60);
  |                           ^^

error: unexpected token: 0
 --> $DIR/invalid_offset.rs:7:21
  |
7 |     let _ = offset!(0);
  |                     ^

error: missing component: sign
 --> $DIR/invalid_offset.rs:8:13
  |
8 |     let _ = offset!();
  |             ^^^^^^^^^
  |
  = note: this error originates in the macro `offset` (in Nightly builds, run with -Z macro-backtrace for more info)

error: invalid component: hour was 0a
 --> $DIR/invalid_offset.rs:9:22
  |
9 |     let _ = offset!(+0a);
  |                      ^^

error: invalid component: minute was 0a
  --> $DIR/invalid_offset.rs:10:24
   |
10 |     let _ = offset!(+0:0a);
   |                        ^^

error: invalid component: second was 0a
  --> $DIR/invalid_offset.rs:11:27
   |
11 |     let _ = offset!(+0:00:0a);
   |                           ^^
